### PR TITLE
Refractor(style?): Simplify AI::Step

### PR DESCRIPTION
**Pull request by RisingLeaf**
## Summary
AI::Step is the most important part of the AI code, but it is very crowded, many things happen directly inside it.

This PR attempts to bring a bit more order into this mess, I created a few more variables, shortened(not in a bad way I hope) some passages and created two new functions for two lengthy bits.

I am opening this as a draft because I first want some opinions if the changes I did are actually good for readability.

## Testing Done
I did not change code, just moved/stylize? it.

## Performance Impact
N/A

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -2066,6 +1899,193 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
...
+	const bool findNewParent = isYours ? !Random::Int(30) : !Random::Int(1800);
+	bool parentHasSpace = inParentSystem && parent->BaysFree(ship->Attributes().Category());
+	if(findNewParent && parentHasSpace && isYours)
+		parentHasSpace = parent->CanCarry(*ship);
+	if(!hasParent || (!inParentSystem && !ship->JumpNavigation().JumpFuel()) || (!parentHasSpace && findNewParent))
+	{
+		// Find the possible parents for orphaned fighters and drones.
+		vector<shared_ptr<Ship>> parentChoices{};
+		parentChoices.reserve(ships.size() * .1);
+		auto getParentFrom = [&ship, &gov, &parentChoices, &system](const list<shared_ptr<Ship>> &otherShips) -> shared_ptr<Ship>
```
Comment (RisingLeaf) [inline source/AI.cpp:1922]:
```suggestion
		auto getParentFrom = [&ship, &gov, &parentChoices, &system](const list<shared_ptr<Ship>> &otherShips)
			-> shared_ptr<Ship>
```

File: source/AI.cpp [hunk 2]
```diff
@@ -46,6 +46,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <algorithm>
 #include <cmath>
 #include <limits>
+#include <memory>
```
Comment (RisingLeaf) [inline source/AI.cpp:49]:
```suggestion
```

File: source/AI.cpp [hunk 3]
```diff
@@ -549,19 +549,17 @@ void AI::Step(Command &activeCommands)
 	step = (step + 1) & 31;
 	int targetTurn = 0;
 	int minerCount = 0;
-	const int maxMinerCount = minables.empty() ? 0 : 9;
-	bool opportunisticEscorts = !Preferences::Has("Turrets focus fire");
-	bool fightersRetreat = Preferences::Has("Damaged fighters retreat");
-	const int npcMaxMiningTime = GameData::GetGamerules().NPCMaxMiningTime();
+	const bool playerOpportunisticEscorts = !Preferences::Has("Turrets focus fire");
+	const bool playerFightersRetreat = Preferences::Has("Damaged fighters retreat");
```
Comment (quyykk) [inline source/AI.cpp:553]: I  think all these variable renames are useless. The names aren't bad enough to justify it imo.

File: source/AI.cpp [hunk 4]
```diff
@@ -549,19 +549,17 @@ void AI::Step(Command &activeCommands)
...
-			continue;
-		// Skip any carried fighters or drones that are somehow in the list.
-		if(!it->GetSystem())
+		const System *system = it->GetSystem();
+
+		// A destroyed or carried ship can't do anything.
+		if(it->IsDestroyed() || !system)
 			continue;
 
+		// The player is their own AI.
```
Comment (quyykk) [inline source/AI.cpp:562]: I don't understand this comment.

File: source/Ship.h [hunk 5]
```diff
@@ -209,6 +209,7 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Set the commands for this ship to follow this timestep.
 	void SetCommands(const Command &command);
 	void SetCommands(const FireCommand &firingCommand);
```
Comment (quyykk) [inline source/Ship.h:211]: Turns out SetCommands with just FireCommand is never used alone, so we can just remove it.

File: source/AI.cpp [hunk 6]
```diff
@@ -2066,6 +1898,194 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
...
+	// If we get here, it means that the ship has not decided to return
+	// to its mothership. So, it should continue to be deployed.
+	command |= Command::DEPLOY;
+
+	return false;
+}
+
+
+
+bool AI::HandlePersonalityTraits(const shared_ptr<Ship> &ship, shared_ptr<Ship> &parent, Command &command,
```
Comment (quyykk) [inline source/AI.cpp:2004]: I'm not too sure about this function since some personality things are still done in AI::Step.

**Review by quyykk — COMMENTED** [review]
I think it's a good idea

**Comment by RisingLeaf** [comment]
No reaction on this, closing it because there apparently is no interest .

**Pull request closed without merging**